docs/extensions.md

```diff
@@ -0,0 +1,308 @@
+
+```bash
+cat <<EOF | kubectl apply -n tekton-pipelines -f -
+kind: Service
+apiVersion: v1
+metadata:
+  name: sample-extension
+  labels:
+    app: sample-extension
+    tekton-dashboard-extension: "true"
+  annotations:
+    tekton-dashboard-display-name: Hello
+    tekton-dashboard-endpoints: sample
+spec:
+  ports:
+  - port: 3000
+    targetPort: 3000
+  selector:
+    app: sample-extension
+---
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: sample-extension
+  labels: 
+    app: sample-extension
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      app: sample-extension
+  template:
+    metadata:
+      labels:
+        app: sample-extension
+    spec:
+      containers:
+      - name: node
+        image: node:latest
+        ports:
+        - containerPort: 3000
+        command:
+        - bash
+        args:
+        - -c
+        - |
+          cat <<EOF > server.js
+          const express = require('express');
+          const app = express();
+          app.get('/sample', (req, res) => res.send('Hello Tekton Dashboard !'));
+          app.listen(3000, '0.0.0.0');
+          EOF
+          npm install express
+          node ./server.js
+EOF
+```
+
+The command above does the following:
+- create a deployment where the `express` web server is installed in a `nodejs` container
+- start the web server and serve the `/sample` endpoint returning the `'Hello Tekton Dashboard !'` message when hit
+- create a service exposing the deployment
+  - the `tekton-dashboard-extension: "true"` label lets the Tekton Dashboard know about the service based extension
+  - the  `tekton-dashboard-display-name: Hello` annotation make the extension appear in the side nav under the `Hello` name
+  - the `tekton-dashboard-endpoints: sample` annotation allows proxying requests to the extension by the Tekton Dashboard
+
+You can verify that the extension backend is working by hitting the path `/v1/extensions/sample-extension/sample`, it should display the `Hello Tekton Dashboard !` message.
+
+![Service based extension backend](./extensions-service-based-backend.png)
+
+The error message happens because the extension doesn't have frontend code yet, [next step](#example-add-and-serve-frontend-code) will guide you through adding frontend code to your extension.
+
+### Example: Add and serve frontend code
+
+The Tekton Dashboard is developed using [React](https://reactjs.org/), therefore the extension frontend code will need to use React too.
+
+All you need to do is provide a js module exporting your `Component`, this component will be loaded at runtime by the Tekton Dashboard host and injected into the page.
+
+For this to work, you will need to create your component using the shared objects provided by the host, namely you won't `import React, { Component } from 'react';` but use `window.React` instead.
```

Comment:
Maybe add something like:
> This can be done manually or using an ES module bundler such as [Rollup](https://rollupjs.org/) to transform your imports.

Reply:
fixed